src/components/VideoProvider/useBackgroundSettings/useBackgroundSettings.ts

```diff
@@ -65,13 +68,61 @@ export const backgroundConfig = {
   images,
 };
 
-// TODO : Add video processing logic after backgroundSettings change
-// useEffect hooks, etc ...
+const virtualBackgroundAssets = '/virtualbackground';
 
-export default function useBackgroundSettings() {
-  const [backgroundSettings, setBackgroundSettings] = useState({
-    type: 'none',
-    index: 0,
-  } as BackgroundSettings);
-  return [backgroundSettings, setBackgroundSettings] as const;
+let blurProcessor: GaussianBlurBackgroundProcessor;
+
+export default function useBackgroundSettings(videoTrack: LocalVideoTrack) {
+  const [backgroundSettings, setBackgroundSettings] = useState<BackgroundSettings>(() => {
+    const localStorageSettings = window.localStorage.getItem(BG_SETTINGS_KEY);
+    return localStorageSettings ? JSON.parse(localStorageSettings) : { type: 'none', index: 0 };
+  });
+
+  // load processors on initialization
+  useEffect(() => {
+    if (!blurProcessor) {
+      blurProcessor = new GaussianBlurBackgroundProcessor({
+        assetsPath: virtualBackgroundAssets,
+      });
+      blurProcessor.loadModel();
+    }
+  }, []);
+
+  // for reapplying background replacement after camera is toggled on again
+  useEffect(() => {
+    const storedBackgroundSettings = window.localStorage.getItem(BG_SETTINGS_KEY);
+    if (videoTrack && storedBackgroundSettings) {
+      const parsedSettings = JSON.parse(storedBackgroundSettings!);
+      if (parsedSettings.type === 'blur') {
+        videoTrack.addProcessor(blurProcessor);
+      }
+    }
+  }, [videoTrack]);
+
+  // update the backgroundSettings and save into localStorage
+  const updateSettings = (settings: BackgroundSettings): void => {
+    setBackgroundSettings(settings);
+    window.localStorage.setItem(BG_SETTINGS_KEY, JSON.stringify(settings));
+  };
+
+  const updateBackgroundSettings = useCallback(
+    async (settings: BackgroundSettings) => {
+      // check if new background settings have been selected
+      if (settings.type !== backgroundSettings.type) {
+        // remove processor
+        if (videoTrack.processor) {
```

Comment:
Just tested and this throws an error if I disable my video. Maybe update this to `if (videoTrack && videoTrack.processor)`?